### PR TITLE
log: Add upgrade progress information to the logs

### PR DESCRIPTION
**Pull request by ystaticy**
### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tidb/issues/58255

Problem Summary:

### What changed and how does it work?
Print progress information during the upgrade process:
```
[2024/12/13 18:13:31.057 +08:00] [INFO] [bootstrap.go:1565] ["upgrade in progress, a version has just been completed or be skipped."] [keyspaceName=ks1] [old-start-version=182] [in-progress-version=upgradeToVer210] [latest-version=239]
```

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Upgrading TiDB from a lower version to a higher version, with  upgrade version update. The following logs can be observed during the process:
```
[2024/12/13 18:13:31.057 +08:00] [INFO] [bootstrap.go:1565] ["upgrade in progress, a version has just been completed or be skipped."] [keyspaceName=ks1] [old-start-version=182] [in-progress-version=upgradeToVer210] [latest-version=239]
```

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
Add upgrade progress information to the logs
```

### CODE REVIEW

**Review by zimulala — APPROVED** [review]
LGTM

**Review by djshow832 — APPROVED** [review]

File: pkg/session/bootstrap.go [hunk 1]
```diff
@@ -802,6 +804,12 @@ func bootstrap(s sessiontypes.Session) {
 	}
 }
 
+func getFunctionName(f func(sessiontypes.Session, int64)) string {
+	fullName := runtime.FuncForPC(reflect.ValueOf(f).Pointer()).Name()
```
Comment (bb7133) [inline pkg/session/bootstrap.go:808]: Even `getFunctionName` is used for `upgrade`s only for now, IMO it's better to add some protections like `nil` and size check. WDYT?

**Review by yudongusa — COMMENTED** [review]
@ystaticy any test to cover this change?

**Comment by ystaticy** [comment]
> @ystaticy any test to cover this change?

@yudongusa I added a UT case,PTAL~

**Review by bb7133 — APPROVED** [review]
LGTM

**Review by benmeadowcroft — APPROVED** [review]

**Pull request merged**